dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -153,12 +153,22 @@
    * @return method and signature entries defined in the class file
    * @throws IOException when there is a problem in reading classFileStream
    */
-  public static List<MethodSignature> listDeclaredMethods(
+  static List<MethodSignature> listDeclaredMethods(
       InputStream classFileStream, String fileName) throws IOException {
-    final ClassParser parser = new ClassParser(classFileStream, fileName);
-    final JavaClass javaClass = parser.parse();
-    Method[] methods = javaClass.getMethods();
+    ClassParser parser = new ClassParser(classFileStream, fileName);
+    JavaClass javaClass = parser.parse();
+    return listDeclaredMethods(javaClass);
+  }
 
+  /**
+   * Lists method signatures from the class file. The output corresponds to entries in the
+   * method table in the .class file.
+   *
+   * @param javaClass .class file to search methods
+   * @return method and signature entries defined in the class file
+   */
+  static List<MethodSignature> listDeclaredMethods(JavaClass javaClass) {
```

Comment:
Looking at this now I wonder whether we truly need the MethodSignature and FullyQualifiedMethodSignature classes, or if we can just pass around BCEL Method objects instead.

Reply:
As BCEL Method has its own equal/hashcode logic (on name and its signature), having our own class gives us more flexibility. For example, BCEL method doesn't compare with accessor (private/public).

https://commons.apache.org/proper/commons-bcel/apidocs/org/apache/bcel/classfile/Method.html#hashCode--